src/DataProvider/DoctrineOrmDataProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meilisearch\Bundle\DataProvider;
+
+use Doctrine\Persistence\ManagerRegistry;
+
+final class DoctrineOrmDataProvider implements DataProvider
+{
+    private string $entityClassName;
+    private ManagerRegistry $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    public function setEntityClassName(string $entityClassName): void
+    {
+        $this->entityClassName = $entityClassName;
+    }
```

Comment:
why making this stateful? I'd register this provider as a separate service for each index and pass the class name via constructor